packages/catalog-client/src/CatalogClient.ts

```diff
@@ -353,6 +354,41 @@ export class CatalogClient implements CatalogApi {
     );
   }
 
+  /**
+   * {@inheritdoc CatalogApi.validateEntity}
+   */
+  async validateEntity(
+    entity: Entity,
+    location: string,
+    options?: CatalogRequestOptions,
+  ): Promise<ValidateEntityResponse> {
+    const response = await this.fetchApi.fetch(
+      `${await this.discoveryApi.getBaseUrl('catalog')}/validate-entity`,
+      {
+        headers: {
+          'Content-Type': 'application/json',
+          ...(options?.token && { Authorization: `Bearer ${options?.token}` }),
+        },
+        method: 'POST',
+        body: JSON.stringify({ entity, location }),
+      },
+    );
+
+    if (response.ok) {
+      return {
+        valid: true,
+        errors: undefined,
+      };
+    }
+
```

Comment:
This should probably check for `if (response.status !== 400) { throw await ResponseError.fromResponse(response); }`

Reply:
Done.